### PR TITLE
If a browser field makes browser-resolve ignore an import, replace with {}

### PR DESCRIPTION
**Pull request by bouk**
@nneufeld

### CODE REVIEW

**Comment by bouk** [comment]
for review @lemonmade

File: js/babel-plugin-sprockets-commoner-internal/index.js [hunk 1]
```diff
@@ -25,7 +25,9 @@ if (typeof Object.assign != 'function') {
 
 var fs = require('fs');
 var dirname = require('path').dirname;
+var join = require('path').join;
 var resolve = require('browser-resolve').sync;
+var emptyModule = join(__dirname, 'node_modules', 'browser-resolve', 'empty.js');
```
Comment (lemonmade) [inline js/babel-plugin-sprockets-commoner-internal/index.js]: Maybe we can just check if the resolved path is inside the `browser-resolve` package? Makes me a bit worried to point at a particular file.

File: js/babel-plugin-sprockets-commoner-internal/index.js [hunk 2]
```diff
@@ -171,14 +181,18 @@ module.exports = function (context) {
...
-        default:
-          // Otherwise we just look for the module by referencing its Special Identifier™
+      case "ExpressionStatement":
+        // We just need to know there's a dependency, we can remove it then
+        path.remove();
+        break;
+      default:
+        // Otherwise we just look for the module by referencing its Special Identifier™
+        if (replacement === false) {
+          path.replaceWith(t.objectExpression([]));
```
Comment (lemonmade) [inline js/babel-plugin-sprockets-commoner-internal/index.js]: This is replacing a call expression, right? Probably want to do an empty function here instead.

Comment (bouk) [inline js/babel-plugin-sprockets-commoner-internal/index.js]: what do you mean by empty function? something like 

`console.log(require('stubbed'))` would become `console.log({})`

File: js/babel-plugin-sprockets-commoner-internal/index.js (hunk 1 above)
Comment (bouk) [inline js/babel-plugin-sprockets-commoner-internal/index.js]: Why? It's what empty modules resolve to https://github.com/defunctzombie/node-browser-resolve/blob/a238dd1ca8db9319cdd6b6340358a09f298c9156/index.js#L56

Comment (bouk) [inline js/babel-plugin-sprockets-commoner-internal/index.js]: it's a bit of an ugly hack, but so is basically the whole javascript module system 😛

Comment (lemonmade) [inline js/babel-plugin-sprockets-commoner-internal/index.js]: lol, yeah, I don't feel too strongly about it, just seems like relying on an internal detail of how `browser-resolve` works. KCSO

File: js/babel-plugin-sprockets-commoner-internal/index.js (hunk 2 above)
Comment (lemonmade) [inline js/babel-plugin-sprockets-commoner-internal/index.js]: Sorry, got confused, thought this was replacing stuff like `require('something')()`

**Comment by lemonmade** [comment]
👍

**Pull request merged**